app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -191,6 +188,14 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        consumeFlow(requireComponents.core.store) { flow ->
```

Comment:
nit: same as above, if we add a `observeSearchChanges` function and pass the store it would be relatively painless to unit tests. That's what I did in `BrowserFragmentTest`, but can also be follow up tickets to improve coverage here.